ledger/complete/ledger.go

```diff
@@ -229,10 +236,39 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, trieUpdate *
 
 	state := update.State()
 	l.logger.Info().Hex("from", state[:]).
-		Hex("to", newRootHash[:]).
+		Hex("to", newState[:]).
 		Int("update_size", update.Size()).
 		Msg("ledger updated")
-	return ledger.State(newRootHash), trieUpdate, nil
+	return newState, trieUpdate, nil
+}
+
+func (l *Ledger) set(trieUpdate *ledger.TrieUpdate) (newState ledger.State, err error) {
+
+	resultCh := make(chan error, 1)
+
+	trieCh := make(chan *trie.MTrie, 1)
+	defer close(trieCh)
```

Comment:
better to close the resultCh as well with `defer`

Reply:
> better to close the resultCh as well with `defer`

No, `Ledger.Set` shouldn't close `resultCh` with `defer`.  

`Ledger.Set` receives from `resultCh` so it shouldn't close the channel.  It may not happen in this case, but only sender should close channel if needed to avoid `panic` :lady_beetle:   when sending to a closed channel.

On the other hand, `Ledger.Set` sends to `trieCh` so it can close the channel.